src/main/java/dev/emi/trinkets/api/TrinketsAttributeModifiersComponent.java

```diff
@@ -0,0 +1,98 @@
+package dev.emi.trinkets.api;
+
+import com.google.common.collect.ImmutableList;
+import com.mojang.serialization.Codec;
+import com.mojang.serialization.codecs.RecordCodecBuilder;
+
+import java.util.List;
+import java.util.Optional;
+
+import net.minecraft.component.DataComponentType;
+import net.minecraft.entity.attribute.EntityAttribute;
+import net.minecraft.entity.attribute.EntityAttributeModifier;
+import net.minecraft.network.RegistryByteBuf;
+import net.minecraft.network.codec.PacketCodec;
+import net.minecraft.network.codec.PacketCodecs;
+import net.minecraft.registry.Registries;
+import net.minecraft.registry.RegistryKeys;
+import net.minecraft.registry.entry.RegistryEntry;
+
+public record TrinketsAttributeModifiersComponent(List<Entry> modifiers, boolean showInTooltip) {
```

Comment:
It replaces reading attributes from nbt, as ItemStacks use components now (also as a bonus, it works like an api extension as it can be set on items by default)